### DISCUSSION TITLE
How do we code $step->isAhead() and $step->isBehind()

### DISCUSSION THREAD

**Discussion by damms005** [spatie/laravel-livewire-wizard#24] [Ideas]
Currently, we thankfully have [these step methods](https://spatie.be/docs/laravel-livewire-wizard/v1/usage/rendering-navigation#content-available-methods):

```
isCurrent(): returns true if this step is currently being displayed
isPrevious(): returns true if this step comes before the step that's currently displayed
isNext(): returns true if this step comes after the step that's currently displayed
show(): return the string that can be passed to wire:click to show the step
```
This assumes a 3-step process. However, for a wizard with like 5 steps, there is a need for more information especially for css styling, such that we can have the following, too: 

```
isAhead(): returns true if this step is ahead of the step that's currently displayed
isBehind(): returns true if this step is behind the step that's currently displayed
```

PRs accepted?

**Comment by damms005** [comment]
Actually, after looking at the source, these new additions should better be properties (so we don't complicate the `StepStatus` enum)

I will push my changes to a fork in a moment. @freekmurze please let me know if this addition will be welcomed for this package, so I can submit a PR from my feature fork

**Reply by freekmurze** [reply]
If the code is simple, well documented and tested, I'd consider pulling in a PR that adds this.

**Comment by damms005** [comment]
@freekmurze I am unable to determine the best way to test this feature. (good thing I have a subscription for [your Laravel Testing course](https://spatie.be/courses/testing-laravel-with-pest)). I have added the test shell. I will appreciate of you can pair-program this with me. Perhaps we can add it to [the YouTube playlist series](https://www.youtube.com/playlist?list=PLSGjn8vP-xYs3Rs9773C-3BkPx70JSy0e)?

I am sure what I will learn from this will be a good addition to what I have already learnt from the [Testing Laravel](https://spatie.be/courses/testing-laravel-with-pest) course

I can thereafter add the documentation

**Reply by freekmurze** [reply]
Thanks for the offer, but I don't have any time to dedicate to this.

**Comment by ju5t** [comment]
I'm not really sure what `isAhead` and `isBehind` would add. Let's say you have a wizard of 5 steps. You're at step 3.

1. `isPrevious` = true
2. `isPrevious` = true
3. `isCurrent` = true
4. `isNext` = true
5. `isNext` = true

So what you're asking is already there. `isNext` and `isPrevious` can be used on all steps, not just step 2 and 4 in this example.
